platform/linuxbsd/wayland/display_server_wayland.cpp

```diff
@@ -642,8 +642,38 @@ void DisplayServerWayland::window_set_title(const String &p_title, DisplayServer
 }
 
 void DisplayServerWayland::window_set_mouse_passthrough(const Vector<Vector2> &p_region, DisplayServer::WindowID p_window_id) {
-	// TODO
-	DEBUG_LOG_WAYLAND(vformat("wayland stub window_set_mouse_passthrough region %s", p_region));
+	ERR_FAIL_MSG("Mouse polygon passthrough not supported by wayland.");
+}
+
+void DisplayServerWayland::window_set_mouse_passthrough_rects(const TypedArray<Rect2i> &p_rectangles, WindowID p_window) {
+	MutexLock mutex_lock(wayland_thread.mutex);
+	main_window.passthrough_rectangles = p_rectangles;
+	_update_window_mouse_passthrough(p_window);
+}
+
+void DisplayServerWayland::_update_window_mouse_passthrough(WindowID p_window) {
+	wl_surface *surface = wayland_thread.window_get_wl_surface(p_window);
+	WaylandThread::WindowState *window_state = wayland_thread.wl_surface_get_window_state(surface);
+
+	if (main_window.flags & (1 << WINDOW_FLAG_MOUSE_PASSTHROUGH)) {
+		wl_region *region = wl_compositor_create_region(window_state->registry->wl_compositor);
+		wl_surface_set_input_region(surface, region);
```

Comment:
I was confused as well when I started working on this. The way passthrough is handled:

- If `WINDOW_FLAG_MOUSE_PASSTHROUGH` is set all other passthrough settings are ignored and we set a full passthrough. We achieve this by creating and setting an empty region
- If  `WINDOW_FLAG_MOUSE_PASSTHROUGH` is not set and there are no passthrough regions (polygons/rects) defined, passthrough is disabled. This is where the infinite input region is used, I'll change the nullptr to NULL but behavior is the same 
- otherwise we create the passthrough region with the rectangles